logfire/_internal/config_params.py

```diff
@@ -183,6 +183,13 @@ def _cast(self, value: Any, name: str, tp: type[T]) -> T | None:
             return value
         if get_origin(tp) is Literal:
             return _check_literal(value, name, tp)
+        if get_origin(tp) is Union:
+            for arg in get_args(tp):
+                try:
+                    return self._cast(value, name, arg)
+                except LogfireConfigError:
+                    pass
+            raise LogfireConfigError(f'Expected {name} to be one of {get_args(tp)}, got {value!r}')
```

Comment:
Seems like this is a great place to use `pydantic` - why don't we?

Reply:
Reducing deps perhaps?

Reply:
> Reducing deps perhaps?

Yes. Pydantic is not a dependency of logfire.